### ISSUE TITLE
Change Request: Add api support for timing information 

### ISSUE THREAD

**Issue by Mattstir** [eslint/eslint#16521]
### ESLint version

8.22.0

### What problem do you want to solve?

Currently it is not possible to get the [TimingInformation](https://eslint.org/docs/latest/developer-guide/working-with-rules#per-rule-performance), when using the [Node.js API](https://eslint.org/docs/latest/developer-guide/nodejs-api).

Only CLI supports the output of the TIMING information. Adding this also for the nodejs api would allow for automatic reporting of slow rules. In a big project with a lot of rules (also a lot of custom - quite heavy ones) this would come in quite handy.

### What do you think is the correct solution?

My take on this would be to add a new option to the `ESLint.Options`.
e.g.: `timing: true`

To not break any current usages, changing the report value of e.g.: lintFiles isn't really possible.
Therefore adding a function that allows to get the timing information of the last lintrun would make sense.

e.g.: `eslint.getLatestTimingInformation` 

This returns an array with a following object per rule:
```typescript
{
    name: string, // name of the rule
    time: number, // time the rule consumed in ms
    relative: number // percent of the total consumed time, e.g.: 0.13 for 13%
}
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by snitin315** [comment]
Since we already allow this for CLI, I feel it makes sense to support this with Nodejs API. I'd like to know what other team members think about this.

**Comment by nzakas** [comment]
:+1: If we have the data, we can expose it. The question is: what is the best way to expose it? We can't just add a method to `ESLint` class because the timing method is specific to each call to `lintText()` or `lintFiles()`. In all likelihood, we might need new methods like `lintTextWithTiming()` and `lintFilesWithTiming()` to avoid breaking changes.

This would require an RFC.

**Comment by bmish** [comment]
Interested in this feature.

**Comment by bmish** [comment]
I was imagining all sorts of additional statistics we could calculate about rules (violations, suppressions, time, etc) in: https://github.com/eslint/eslint/issues/14597#issuecomment-1003863524. It would be ideal if this information was available via both CLI and Node API. So it could be worthwhile to account for additional "summary" metrics in any new API.

**Comment by Mattstir** [comment]
Again, Please don't auto-close this issue, as this is still needed! :)

**Comment by nzakas** [comment]
@Mattstir best way to make sure this doesn't get closed is to open an RFC. The core team won't be working on this, so unless someone from the community chips in, this won't get done.

**Comment by fasttime** [comment]
Not stale. Waiting for https://github.com/eslint/rfcs/pull/108.

**Comment by fasttime** [comment]
The RFC has been accepted it is being implemented in #17850.

**Comment by nzakas** [comment]
https://github.com/eslint/eslint/pull/17850 has been merged, so closing.
